uber/models.py

```diff
@@ -630,9 +630,22 @@ def everything(self, location=None):
                                         .order_by(Attendee.full_name).all()
                          if c.AT_THE_CON or not location or int(location) in a.assigned_depts_ints]
             for job in jobs:
-                job._available_staffers = [a for a in attendees if not job.restricted or a.trusted]
+                job._available_staffers = [a for a in attendees if not job.restricted or a.trusted_in(job.location)]
             return jobs, shifts, attendees
 
```

Comment:
> if the new query is way faster [which, it really should be...] (famous last words), then we might not need any of this anymore.

I'm sure it is faster; having looked at your new version, it's definitely going to see some performance improvements.  However, suppose we have a page with all jobs yet to be worked, which can be literally hundreds of jobs, and we want to iterate over them and show all of the available staff (something we actually do).  Without this cache preloading, we're doing a new database query which loads potentially hundreds of attendees for every one of those hundreds of jobs, which is probably not ideal.

``session.everything()`` has definitely hit a lot of growing pains with this stuff, since it dates back to a time when we only had a few dozen jobs and a few hundred attendees in the entire database.  I've put a few performance band aids on it over time, but I've never gone back and looked really thoroughly at all the places where we use this or do something similar to figure out the right functions that need to exist :(

What you've done here is a good step in that direction though.  I definitely like ``.available_staff`` vs ``.capable_staff``.  (Though I'd probably say ``_volunteer`` instead of ``_staff`` since we do have a distinction between volunteers and staffers, haha.)